From ordex@autistici.org Fri Nov 16 16:49:15 2012
X-Apparently-To: lindner_marek@yahoo.de via 188.125.84.188; Fri, 16 Nov 2012 08:50:20 +0000
Received-SPF: pass (domain of autistici.org designates 178.255.144.35 as permitted sender)
X-YMailISG: LjItQMwWLDsFo9iIj9SpdJB2BfvjP1gD3E_NVQ_UwftNCcRw
 mktGb.BmfJsKNd8xzk6l6Af7UUv1xLGNNu25H2xh2Qb5gkNtOz8STnhREHXW
 2wrVlJrV4N5P1GjUMi.mg0HiNw_iW.wstmwKVt21QjNLaCOFSu8oPvFbkYfI
 a1bHeL4AaHEU0DrQEdLR8RQ74pEv8hqcWNNMJfG53AiFszd5zBkge1YX_Ce.
 tM6HQM.mvYm3leQRAbo9ITvWqlIaaGjSWIinrbKMiIF_MeHSsGZV2vKBn7n6
 q6Afw5YPoHlvIKMLhPDHkCilJ6kL0s7d4pgxe.s9qywWqJjlQUiik_1epdgG
 7EWPJXpL7s_R.rfVjXwyIP.m45MUYOO9yqC_CRpSFikVk4kKhPpALa0dUZOb
 ts_1EEStTcoclQAhj2gkxMrTwwGrV4L.thb3f8pRxEzNf._haOS7.I8EXd6a
 UEqTCSWEFifITimxDl5iXOn1nov6gi5BthoGXOdf1.3ljQcyrzZnMWsplq5D
 u7B_Gjk8YHeWeZma9AhXuCcvmlOBKwNRaGwNy7wI7Bx8KMei7B9cr2jryis3
 jnOGkxmyerkvMvP1e8HQrMC.ge0q718RWZJB_U5JsyNqcCEr7W4qyOggbpcw
 E5igrmVqqs2YbDi5LC7.8vOqlVNKrOvfZpGK4BYDU8eVxCNv2yd40Chd2a94
 xUqzzSCq_SDsEvheC1lxOWIEFh3OXjYjp3bx3Eugx1.9gWDOX2k5BLpKkg.A
 IM9CcDNqLPdfsQcPCoapHq6vOumxN9pZ8JxiNKRbDmy25ilJlOARJ9rcYTvA
 lYH.22GFKzx3ggAuW89UVsw0agTkF55kT.9_84nTSfmle5cUwO5t4lEFXhq9
 V5hy3rjYs8FShSBR6UC5rav23vPHsQESjqLCE0Q4H4HYA._lGDd4qAErDqwe
 lToZWLoLTW7telXC2LTnC3JeG8J7L4wJGo9PbfbPx6En.bj6GgfY5qObT0bk
 q.5ZOl_vfdcdnjw5sVpMuxpIzRyASOM4Dp40Kd5VF6zajUeGuRA8bXs34Ag-
X-Originating-IP: [178.255.144.35]
Authentication-Results: mta1083.mail.ukl.yahoo.com  from=autistici.org; domainkeys=neutral (no sig);  from=autistici.org; dkim=pass (ok)
Received: from 127.0.0.1  (EHLO contumacia.investici.org) (178.255.144.35)
  by mta1083.mail.ukl.yahoo.com with SMTP; Fri, 16 Nov 2012 08:50:19 +0000
Received: from [178.255.144.35] (contumacia [178.255.144.35]) (Authenticated sender: ordex@autistici.org) by localhost (Postfix) with ESMTPSA id 00F60E8714;
	Fri, 16 Nov 2012 08:50:17 +0000 (UTC)
X-DKIM: Sendmail DKIM Filter v2.8.2 contumacia.investici.org 00F60E8714
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=autistici.org;
	s=stigmate; t=1353055819;
	bh=XsI1HH9Qj6JfIxg3GZGAztofUiRgq5pmBZUCGlj/oKs=;
	h=From:To:Cc:Subject:Date:Message-Id:In-Reply-To:References;
	b=ahp9/Hf4dpprZKMhl2rN3hK5FGhM61FzXZgxzMdz0KqKTvJfjcaS0Ii0a8pAVFilq
	 TfTcqXNuPDbRE/i6wdU6+QEY1/fkRK7Ij2rWdQECxEVe5pItj8vMCvXH+GYlhjQkcs
	 ys7ZMqyXXzs9dUqZRenQXLJLuTkcSwuPP7bmGsg0=
From: Antonio Quartulli <ordex@autistici.org>
To: davem@davemloft.net
Cc: netdev@vger.kernel.org,
 Simon Wunderlich <simon.wunderlich@s2003.tu-chemnitz.de>,
 Marek Lindner <lindner_marek@yahoo.de>,
 Sven Eckelmann <sven@narfation.org>,
 Antonio Quartulli <ordex@autistici.org>
Subject: [PATCH] batman-adv: fix tt_global_entries flags update
Date: Fri, 16 Nov 2012 09:49:15 +0100
Message-Id: <1353055758-2901-2-git-send-email-ordex@autistici.org>
X-Mailer: git-send-email 1.8.0
In-Reply-To: <1353055758-2901-1-git-send-email-ordex@autistici.org>
References: <1353055758-2901-1-git-send-email-ordex@autistici.org>
Status: R
X-Status: N
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  

Flags carried by a change_entry have to be always copied into the
client entry as they may contain important attributes (e.g.
TT_CLIENT_WIFI).

For instance, a client added by means of the "early detection
mechanism" has no flag set at the beginning, so they must be updated once the
proper ADD event is received.

This was introduced by 30cfd02b60e1cb16f5effb0a01f826c5bb7e4c59
("batman-adv: detect not yet announced clients")

Signed-off-by: Antonio Quartulli <ordex@autistici.org>
Signed-off-by: Marek Lindner <lindner_marek@yahoo.de>
---
 translation-table.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/translation-table.c b/translation-table.c
index 112edd3..64c0012 100644
--- a/translation-table.c
+++ b/translation-table.c
@@ -769,6 +769,12 @@ int batadv_tt_global_add(struct batadv_priv *bat_priv,
 		 */
 		tt_global_entry->common.flags &= ~BATADV_TT_CLIENT_TEMP;
 
+		/* the change can carry possible "attribute" flags like the
+		 * TT_CLIENT_WIFI, therefore they have to be copied in the
+		 * client entry
+		 */
+		tt_global_entry->common.flags |= flags;
+
 		/* If there is the BATADV_TT_CLIENT_ROAM flag set, there is only
 		 * one originator left in the list and we previously received a
 		 * delete + roaming change for this originator.
-- 
1.8.0

